public/modules/core/controllers/home.client.controller.js

```diff
@@ -83,6 +88,29 @@ angular.module('core').controller('HomeController', ['$scope', '$location', 'Aut
                 }
             });
 
+            $location.search('country', ['United States']);
+        }
+
+
+
+        //search in the mapping table
+        function searchMappingByStatus() {
+
+            Mappings.searchByStatus.query({
+                    status: true
+                },
+                function (data) {
+                    if (data.length > 0) {
+                        for (var i = 0; i < data.length; i++) {
+                            $scope.comTrialIds.push(data[i].nctId);
+                        }
```

Comment:
It's not recommend to push the item into scope in a for loop. It will trigger digest loop every time. Try to push to a local array then sigh it to scope